source/RemoteWiring/HardwareProfile.h

```diff
@@ -0,0 +1,351 @@
+    in the Software without restriction, including without limitation the rights
+    to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+    copies of the Software, and to permit persons to whom the Software is
+    furnished to do so, subject to the following conditions :
+
+    The above copyright notice and this permission notice shall be included in
+    all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+    THE SOFTWARE.
+*/
+
+#pragma once
+
+namespace Microsoft {
+namespace Maker {
+namespace RemoteWiring {
+
+/*
+ * Protocol enum is used to recognize which protocol the initialization data represents.
+ * Currently, the only option is Firmata, which is the only protocol Remote Arduino currently supports
+ */
+public enum class Protocol
+{
+    FIRMATA
+};
+
+/*
+ * Pin capabilities are stored as bitmasks, the PinCapability enum represents the bit value of each capability.
+ */
+public enum class PinCapability
+{
+    INPUT = 0x01,
+    INPUT_PULLUP = 0x02,
+    OUTPUT = 0x04,
+    ANALOG = 0x08,
+    PWM = 0x10,
+    SERVO = 0x20,
+    I2C = 0x40
+};
+
+/*
+ * This class represents a virtual piece of hardware. It can create a profile of pins and their capabilities, and can be
+ * used to verify outgoing commands are valid and map incoming commands to specific pins.
+ */
+public ref class HardwareProfile sealed
+{
+
+public:
+
+    //this is a required property which must always be accurate
+    property int AnalogPinCount
+    {
+        int get()
+        {
+            return _num_analog_pins;
+        }
+    }
+
+    //this is a required property which must always be accurate
+    property int AnalogOffset
+    {
+        int get()
+        {
+            return _analog_offset;
+        }
+    }
+
+    //this is a required property which must always be accurate
+    property bool IsValid
+    {
+        bool get()
+        {
+            return _is_valid;
+        }
+    }
+
+    //this is a required property which must always be accurate
+    property int TotalPinCount
+    {
+        int get()
+        {
+            return _total_pins;
+        }
+    }
+
+    property Windows::Foundation::Collections::IVector<uint8_t> ^AnalogPins
+    {
+        Windows::Foundation::Collections::IVector<uint8_t> ^ get()
+        {
+            auto vector = ref new Platform::Collections::Vector<uint8_t>();
+            if( _is_valid )
+            {
+                for( uint8_t pin = _analog_offset; pin < _total_pins; ++pin )
```

Comment:
I understand the desire to save space and only use `uint8_t`, but this comes at a performance cost. I think it would be best to use a generic `size_t` because it maps to `unsigned int` which is the optimal size for the machine.